### ISSUE TITLE
import error on windows machines

### ISSUE THREAD

**Issue by wenbowangnrel** [GMLC-TDC/HELICS#1529]
**Describe the bug**
ImportError: DLL load failed: The specified module could not be found.
![image](https://user-images.githubusercontent.com/58447573/90271662-664b9700-de19-11ea-84e8-ce050327a036.png)

**Environment (please complete the following information):**

- Operating System: [Windows]
- what compiler or setup process did you use[Anaconda Prompt - python]
- HELICS version: 

```python
import helics
helics.helicsGetVersion()
'2.5.2 (2020-06-14)'
```

**Additional context and information**
after installing the latest Visual C++ Redistributable for VS 2019, the issue can be solved.

**Comment by nightlark** [comment]
It sounds like after following the suggestion it gave to install VS2019, the problem was fixed? If it was, this issue like a duplicate of https://github.com/GMLC-TDC/HELICS/issues/1139#issuecomment-602719129

The suggested fix message is because almost every time that "DLL load failed" message appears, the problem is the runtime library on your system isn't new enough (the original error message of just "DLL load failed" is very uninformative). If following the suggestion made it work, then the suggestion message that was added did its job, and you shouldn't need to worry about the error appearing again on that computer unless you do something like reinstalling the OS.

This fix has also been mentioned in these places on the docs:
- https://docs.helics.org/en/latest/installation/package_manager.html#troubleshooting-shared-library-errors-on-windows
- https://github.com/GMLC-TDC/HELICS#windows

We don't bundle the VS2019 runtime DLL with HELICS because there can be issues on Windows if different versions of a shared library with the same are loaded into a process at the same time (which could be more annoying to pinpoint as the issue and fix), and it's a 1 time fix for users to update their system.

If you're still having issues after installing the updated VS2019 runtime, then run the tool mentioned in https://github.com/GMLC-TDC/HELICS/issues/1139#issuecomment-602721539 on the dll files that are bundled in the HELICS Python package to provide more information on exactly what library is not found.

**Comment by wenbowangnrel** [comment]
yes, after installing the latest Visual C++ Redistributable for VS 2019, the issue was solved. 
It indeed looks like a issue that is duplicate of #1139 after checking the link. 

The full screen is like this:
![image](https://user-images.githubusercontent.com/58447573/90298430-8007d100-de4f-11ea-8630-c1741bf877bf.png)

Thank you for the explanation.

**Comment by kdheepak** [comment]
@nightlark is there a way to identify that the visual c++ redistributable is not installed? I can add checks and print out a error message accordingly in the new python helics API. 

This also isn't a problem when using mingw compilers on Windows. Thoughts on using that for the official distribution?

**Comment by nightlark** [comment]
I didn't see any way to identify which library it isn't finding. I think Python is just passing along an error message that Windows gives it, since I've seen "The specified module could not be found" in other contexts as well -- it's a very uninformative error message. I suppose if the error happens you could try to guess some locations on the Windows search path and see if you find `vcruntime140_1.dll` and then give a more/less strongly worded suggestion for fixing the problem?

The mingw compilers could have the same problem with conflicting libraries of the same name, except we'd be forced to bundle the mingw DLLs with HELICS.

**Comment by nightlark** [comment]
The only times this error has come up is that one missing system runtime library -- https://github.com/GMLC-TDC/helics-packaging/blob/master/helics-pip/helics/__init__.py has the code that catches the error on Windows and prints the suggested fix (Dheepak).

I don't like how static linking (and bundling DLLs) makes it hard to update the runtime libraries independent of HELICS and adds to the size (mainly for people who have a system with the latest runtime already and maybe update it more often), but since this issue seems to come up every few months, maybe we should switch to static linking the MSVC runtime on Windows.
